### ISSUE TITLE
New Version Release

### ISSUE THREAD

**Issue by krish-adi** [krish-adi/barfi#36]

**Comment by zabrewer** [comment]
@krish-adi - first I hope you are well.

Are the new changes going to break the API or way that barfi is called?  Do you expect any breaking changes?

Will streamlit still be used?

I have a project I'm close to releasing that uses barfi as a dependency.

**Comment by krish-adi** [comment]
Hey @zabrewer ! 

The API will not change, and will still use Streamlit. 

What will mostly change in the upcoming release is the UI. But the blocks and their interaction in the UI will remain the same.

**Comment by krish-adi** [comment]
I will mark a few issues as enhancements, that I'll look into deploying into the next release.

**Comment by equidna** [comment]
Could you add support for jupyterlab? It would be incredible to manage the functions workflow inside a jupyter notebook (maybe something similar to makinarocks link but FOSS).

**Comment by krish-adi** [comment]
hey @equidna , a jupyterlab extension is something I have been thinking about, so will put it in the roadmap. I haven't heard about Makinarocks, could you share an image and a description of what you are referring to and/or expecting to get?

**Comment by equidna** [comment]
@krish-adi The makinarocks link allows to use any notebook cell as a node (link with images: https://link.makinarocks.ai/). Unfortunately the code is closed source and the user needs to create an account to use it.

There is another similar solution called elyra pipelines (link with images: https://elyra.readthedocs.io/en/latest/getting_started/overview.html#ai-pipelines-visual-editor). This tool allows to use any notebook as a node, but is hard to schedule pipeline runs without kubernets or apache airflow and I think it doesn't have an updated version for jupyterlab 4.

My idea would be to run barfi as the output of a cell and instantiate functions of a notebook or cell as nodes (allowing for easier prototyping and testing different workflows). I believe this would allow to parameterize code workflows and code reuse. Instead of just parameterize input values, like the usual papermill solution (https://github.com/nteract/papermill).

**Comment by ANGJustinl** [comment]
Could you add groups setting, like categorize nodes into diffrent groups?

**Comment by grassjelly** [comment]
is the current branch useable? Would love to try and provide feedback

**Comment by krish-adi** [comment]
Hey @grassjelly , the branch is usable, you could check out the app in tests/v1_test.py for the working version.

**Comment by grassjelly** [comment]
Good stuff. Thanks for releasing this. Will the new version support cyclic connections?

**Comment by krish-adi** [comment]
@grassjelly , yes indeed ! i want to abstract away the compute engine from the streamlit component and strip the dependencies. this would help implement cyclic connections/flows.

**Comment by krish-adi** [comment]
Hey folks, v 1.0.0 is out 🎉  ! 

A bunch of major changes went into this. Specifically all the objects being typed. There have been a few convinience functions added to most of the interactions. Refer to the [Changelog](https://github.com/krish-adi/barfi/releases/tag/1.0.0) for more info on this. 

Plus there is a new website [barfi.ai](https://barfi.ai). There will be a hard focus on AI workflows and creating Blocks to suit these. That said, Barfi is only a framework, and can be used for any type of workflow out there, maintaining the same philosophy as when I started working on this.
